### PR TITLE
urlencoded extractor

### PR DESCRIPTION
**Pull request by bosukas**

### CODE REVIEW

File: libs/pavex/src/request/body/urlencoded.rs [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+    UrlEncodedContentTypeMismatch,
+};
+use crate::request::body::BufferedBody;
+use crate::request::RequestHead;
+use http::HeaderMap;
+use serde::Deserialize;
+
+#[doc(alias = "UrlEncoded")]
+#[derive(Debug)]
+/// Parse the body of an incoming request as an application/x-www-form-urlencoded form.
```
Comment (LukeMathWalker) [inline libs/pavex/src/request/body/urlencoded.rs:12]: Most of this should go in a new guide section under [this one](https://pavex.dev/docs/guide/request_data/body/deserializers/)

Comment (bosukas) [reply libs/pavex/src/request/body/urlencoded.rs:12]: Just to clarify, shall I keep the comments in the code, and add a section under docs for urlencoded?

Comment (bosukas) [reply libs/pavex/src/request/body/urlencoded.rs:12]: I will look at how you have done JsonBody and mimic that one, to make it consistent.

Comment (bosukas) [reply libs/pavex/src/request/body/urlencoded.rs]: @LukeMathWalker how does this seem?

**Comment by LukeMathWalker** [comment]
Can you fill in https://github.com/LukeMathWalker/pavex/blob/main/CONTRIBUTORS.md?

File: docs/guide/request_data/body/deserializers/urlencoded.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# UrlEncoded
+
+[`UrlEncodedBody<T>`][UrlEncodedBody] buffers the body in memory and deserializes it as JSON,
```
Comment (LukeMathWalker) [inline docs/guide/request_data/body/deserializers/urlencoded.md:3]:
```suggestion
[`UrlEncodedBody<T>`][UrlEncodedBody] buffers the body in memory and deserializes it as URL-encoded,
```

File: docs/guide/request_data/body/deserializers/urlencoded.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+Pavex supports this use case—**you can borrow from the request body**.
+
+### Percent-encoding
+
+It is not always possible to avoid allocations when handling an urlencoded body.  
+A urlencoded body must comply with the restriction of the URI specification:
+you can only use [a limited set of characters](https://datatracker.ietf.org/doc/html/rfc3986#section-2).  
+If you want to use a character not allowed in a URI, you
+must [percent-encode it](https://developer.mozilla.org/en-US/docs/Glossary/Percent-encoding).  
+For example, if you want to use a space in a query parameter, you must encode it as `%20`.
```
Comment (LukeMathWalker) [inline docs/guide/request_data/body/deserializers/urlencoded.md:48]:
```suggestion
For example, if you want to use a space in a field name or a field value, you must encode it as `%20`.
```

File: libs/pavex/Cargo.toml [hunk 4]
```diff
@@ -47,6 +47,9 @@ serde_html_form = "0.2"
 serde_json = "1"
 serde_path_to_error = "0.1"
 
+# URL encdoed body extractor
+form_urlencoded = "1.2.1"
```
Comment (LukeMathWalker) [inline libs/pavex/Cargo.toml:51]: We already have `serde_html_form` in our dependency tree, we should use it instead of adding a new dep.

**Comment by LukeMathWalker** [comment]
Yes, let's explain it!

**Pull request merged**